Fix ALTER COLUMN options (#39601)
ydb-platform/ydb: ydb/docs/en/core/yql/reference/syntax/_includes/column_option_list_alter.md

### FAMILY <family_name> (column setting)

Specifies that this column belongs to the specified column group. For more information, see [{#T}](../create_table/family.md).

### DEFAULT <default_value>

{% note warning %}

The `DEFAULT` option is supported:

* Only for [row-oriented](../../../../concepts/datamodel/table.md#row-oriented-tables) tables. Support for [column-oriented](../../../../concepts/datamodel/table.md#column-oriented-tables) tables is under development.
* Only with literal values. Support for computed expressions is under development.

{% endnote %}

Allows you to set a default value for a column. If no value is specified for this column when inserting a row, the specified default value will be used. The default value must match the column's data type.

### NOT NULL

> Disallows `NULL` values in this column.
> 
> {% note warning %}
> 
> Currently only `DROP NOT NULL` is supported for this option — it removes the constraint and allows `NULL` values in the column again. `SET NOT NULL` is under development.
> 
> {% endnote %}

Reviewer: The section opens with "Disallows NULL values" which implies `SET NOT NULL` works, but the very next note says it doesn't. Suggest leading with the restriction to avoid confusion:
```suggestion
### NOT NULL

{% note warning %}

Currently only `DROP NOT NULL` is supported. `SET NOT NULL` is under development.

{% endnote %}

Removes the `NOT NULL` constraint from the column, allowing `NULL` values again.
```

### COMPRESSION([algorithm=<algorithm_name>[, level=<value>]])

{% if oss == true and backend_name == "YDB" %}

{% include [OLAP_only_allow_note](../../../../_includes/only_allow_for_olap_note.md) %}

{% endif %}

You can set the following compression parameters for columns:

* `algorithm` — compression algorithm. Allowed values: `off` (disable compression), `lz4`, `zstd`.

* `level` — compression level; supported only for `zstd` (allowed values are 0 through 22).

> If `COMPRESSION()` is specified without parameters, the column uses the default compression. Currently that is `lz4`; future versions will let you configure default compression at the cluster or table level.

Reviewer: The RU file documents `ENCODING([OFF|DICT])` and the RU syntax template includes `[ENCODING([OFF|DICT])]`. The EN file has neither — the EN syntax references an undocumented option. Please add the ENCODING section (translated from RU):
```suggestion
If `COMPRESSION()` is specified without parameters, the column uses the default compression. Currently that is `lz4`; future versions will let you configure default compression at the cluster or table level.

### ENCODING([OFF|DICT])

{% if oss == true and backend_name == "YDB" %}

{% include [OLAP_only_allow_note](../../../../_includes/only_allow_for_olap_note.md) %}

{% endif %}

Sets the encoding for a column's data.

Available options:

* `ENCODING(DICT)` — enables dictionary encoding. Repeated values are replaced by small integer identifiers stored in a dictionary. Effective for low-cardinality columns (few unique values). Supported only for comparable types such as `String`, `Timestamp`, `UInt64`. Using `ENCODING(DICT)` on non-comparable types (`Json`, `JsonDocument`, `Yson`) returns an error.

* `ENCODING(OFF)` — disables special encoding. Data is stored in the standard format without additional encoding.

If `ENCODING()` is specified without parameters, the column uses the default encoding. Currently that is `OFF`; future versions will let you configure the default encoding at the database or table level.
```

> If `COMPRESSION()` is specified without parameters, the column uses the default compression. Currently that is `lz4`; future versions will let you configure default compression at the cluster or table level.

Reviewer 2: Обрати внимание.

### FAMILY <family_name> (column setting)

Specifies that this column belongs to the specified column group. For more information, see [{#T}](../create_table/family.md).

### DEFAULT <default_value>

{% note warning %}

The `DEFAULT` option is supported:

> * Only for [row-oriented](../../../../concepts/datamodel/table.md#row-oriented-tables) tables. Support for [column-oriented](../../../../concepts/datamodel/table.md#column-oriented-tables) tables is under development.

Reviewer 2: Не надо писать under development, так где по факту у нас конкретных планов нет.

* Only with literal values. Support for computed expressions is under development.

{% endnote %}

Allows you to set a default value for a column. If no value is specified for this column when inserting a row, the specified default value will be used. The default value must match the column's data type.

### NOT NULL

{% note warning %}

> Currently only `DROP NOT NULL` is supported. `SET NOT NULL` is under development.

Reviewer 2: Аналогично, предлагаю просто писать про то что мы поддерживаем.
